### ISSUE TITLE
SQL Always on DSC Failure

### ISSUE THREAD

**Issue by CraigSCCM** [Azure/azure-quickstart-templates#1951]
Hi guys i'm trying to deploy the https://github.com/Azure/azure-quickstart-templates/tree/master/sql-server-2014-alwayson-dsc template, it deploys all my resources but fails on provisioning the SQL VM Extension and i have no idea why :(

````
NewAzureRmResourceGroupDeployment : 
Resource Microsoft.Compute/virtualMachines/extensions 'vm01SqlC0/sqlAOPrepare' failed 
with message 'The resource operation completed with terminal provisioning 
state 'Failed

NewAzureRmResourceGroupDeployment : VM has reported a failure when 
processing extension 'sqlAOPrepare'. Error message: "Errors occurred while 
processing configuration 'PrepareAlwaysOnSqlServer

NewAzureRmResourceGroupDeployment : 
Resource Microsoft.Resources/deployments 'deploySqlAlwaysOn' failed with message 'The 
resource operation completed with terminal provisioning state 'Failed`

```

`VMExtenstionProvisioningError, the VM has reported a failure when processing extension 'sqlAOPrepare' Errors occured while processing configuration 'PrepareAlwaysOnSqlServer' `

**Comment by jrudley** [comment]
New-AzureRmResourceGroupDeployment : 6:04:01 PM - VM has reported a failure when processing extension 'sqlAOPrepare'. Error message: "The VM is running a newer WMF (5) than specified in the WmfVersion parameter: (4, 4.0) Please retry your 
configuration on using the VM's current WMF or newer, or on a VM that has the desired WMF version.".

i'd change "typeHandlerVersion": "2.13", to "typeHandlerVersion": "2.16 and remove the 
	"wmfVersion": "4.0" all together. I haven't tested it, but I believe this will fix it. Will try it out tomorrow

**Comment by CraigSCCM** [comment]
:(

**Comment by jrudley** [comment]
yeah, not sure..looks like this repo hasn't been maintained for some time as well

**Comment by CraigSCCM** [comment]
i get the exact same failure on a different template https://github.com/Azure/azure-quickstart-templates/tree/master/sqlvm-alwayson-cluster

**Comment by singhkay** [comment]
@CraigSCCM We might be deprecating this template soon. Try the newer one that's contributed by the SQL team https://github.com/Azure/azure-quickstart-templates/tree/master/sqlvm-alwayson-cluster and let us know if you have any trouble

cc: @weidaiMS

**Comment by CraigSCCM** [comment]
@singhkay hi mate, i did mention above that i get the exact same error on the other template :(

**Comment by jrudley** [comment]
remove the .txt extension and try provisioning with this one. This seems to be the default template they are using from the portal when you provision that way. This worked for me

[DefaultTemplate.json.txt](https://github.com/Azure/azure-quickstart-templates/files/270479/DefaultTemplate.json.txt)

**Comment by singhkay** [comment]
@CraigSCCM My bad. @weidaiMS Can you look at this?

**Comment by weidaiMS** [comment]
@CraigSCCM  Hi Craig, can you provide your parameter list for your deployment?
